src/noit_message_decoder.c

```diff
@@ -458,15 +459,29 @@ noit_metric_tags_parse_one(const char *tagnm, size_t tagnmlen,
   return tagnm + cur_size;
 }
 
+const char *noit_metric_tags_parse_one(const char *tagnm, size_t tagnmlen,
+                                       noit_metric_tag_t *output,
+                                       mtev_boolean *toolong) {
+  return parse_metric_tag(tagnm, tagnmlen, output, toolong,
+                          NOIT_TAG_MAX_PAIR_LEN);
+}
+
 static int
 tag_canonical_size(noit_metric_tag_t *tag) {
-  int len;
-  char dbuff[NOIT_TAG_MAX_PAIR_LEN];
-  len = noit_metric_tagset_decode_tag(dbuff, sizeof(dbuff), tag->tag, tag->total_size);
-  if(len < 0) return 0;
-  len = noit_metric_tagset_encode_tag(dbuff, sizeof(dbuff), dbuff, len);
-  if(len < 0) return 0;
-  return len;
+  mtev_dyn_buffer_t dbuff;
+  mtev_dyn_buffer_init(&dbuff);
+  size_t max_decode_len = mtev_b64_max_decode_len(tag->total_size);
+  mtev_dyn_buffer_ensure(&dbuff, mtev_b64_encode_len(max_decode_len));
+  int len = noit_metric_tagset_decode_tag((char *)mtev_dyn_buffer_data(&dbuff), max_decode_len, tag->tag, tag->total_size);
+  if(len >= 0){
```

Comment:
clang-format your new/updated code please

Reply:
I undid some formatting because it was inconsistent with the surrounding code. should I format the surrounding code to match or only format my updated lines?

Reply:
this:

> clang-format your new/updated code please